ansible/configs/ocp-workshop/files/hosts_template.3.10.45.j2

```diff
@@ -308,6 +308,26 @@ template_service_broker_install=true
 ansible_service_broker_install=true
 ansible_service_broker_local_registry_whitelist=['.*-apb$']
 
+###########################################################################
+### Cloud Provider Specific Vars
+###########################################################################
+
+{% if cloud_provider == 'ec2' %}
+{% elif cloud_provider == 'azure' %}
+openshift_storageclass_parameters={'kind': 'managed', 'storageaccounttype': 'Premium_LRS'}
+openshift_cloudprovider_kind=azure
+openshift_cloudprovider_azure_client_id="{{ azure_service_principal }}"
```

Comment:
Can we create a service principal on-the-fly when provisioning, and then use it here instead of our main SP ?
This way, if a user has SSH access to the environment, he/she cannot use it once the environment is gone.

Reply:
Yes, I can build that into the role for creation and then add it to the destroy_env role as well

Reply:
I'll reuse the same service principal in the install osba role

Reply:
OK, I've been fighting with this off and on for the last month and I don't think it is possible to have a new account created via automation as a service principal. You need to be logged in as a regular user account. I've asked Microsoft and internal Red Hat people and no one has a way to do it.

Reply:
waw.. thanks for looking into this for so long.

Just out of curiosity, why can't we simply do:

```az ad sp create-for-rbac```

?

Reply:
To do automation they recommend a service principal be used and that is how all the scripts we have are setup.
If you run that command as a normal user account with Owner access on the subscription it will work.
If you run that command as a service principal, even if the role is Owner, it returns an error about not enough access, and for every doc I read and every person I talk to, there doesn't seem to be a way to do it.